storage/integration_test.go

```diff
@@ -2438,9 +2439,17 @@ func TestIntegration_WriterContentType(t *testing.T) {
 				setType:  "image/jpeg",
 				wantType: "image/jpeg",
 			},
+			{
+				// Content type sniffing disabled.
+				content:                "<html><head><title>My first page</title></head></html>",
+				forcedEmptyContentType: true,
```

Comment:
I think this is missing `setType: ""` and one of (`wantType: "application/octet-stream"` or `wantType: "application/x-www-form-urlencoded"`), per [GCS documentation](https://cloud.google.com/storage/docs/metadata#content-type): 
![image](https://github.com/googleapis/google-cloud-go/assets/1907648/d6a1ee9b-04cb-4df4-bfab-cd292768098e)

Reply:
We want `setType` to be `""` and `wantType `to be `""`. We actually don't want`application/octet-stream`, since this makes it possible to override the value with `Response-Content-Type` later.

Reply:
Interesting, I've never come across this option.

Does this option work for you after this? I wasn't able to see the additional header Response-Content-Type as it takes the default. I changed the contentType using Cloud Storage browser so that might be why but want to confirm.

![image](https://github.com/googleapis/google-cloud-go/assets/1907648/16f84fae-a13a-47f8-879a-9163cd165022)

Reply:
Yes, `Response-Content-Type` won't work unless the object metadata is blank; this is why we keep it blank. Try clearing the metadata to see.

Reply:
It's also possible that `Response-Content-Type` only works in signed URLs. That's what we use.